library/src/schemas/date/date.ts

```diff
@@ -60,7 +60,7 @@ export function date(
      */
     _parse(input, info) {
       // Check type of input
-      if (!(input instanceof Date)) {
+      if (!(input instanceof Date) || Number.isNaN(input.getTime())) {
```

Comment:
Can you check if `isNaN(…)` also works to save some bytes?

Reply:
It does. I was just following the  number schema example.

Reply:
Theoretically, `Number.isNaN(...)` is safer because it does not transform the value into a number. However, in both cases it makes no difference because we already know that the value is a number or `NaN`. So it does not matter how we do it. We should just be consistent. What do you think is better?

Reply:
I agree. I think because we know it's a number or NaN, isNaN is better to save bytes. I have added a commit to the PR.